records/2023-01-24.malicious-content-scanning.md

```diff
@@ -0,0 +1,61 @@
+# Malicious Content Scanning
+
+Date: 2023-01-24
+
+## Status
+
+**DRAFT**.
+
+_VALUES: (DRAFT, IN REVIEW, APPROVED, REJECTED)_
+
+## Context
+
+We are working on detecting malicious content in attached files, as described in [this issue card](https://app.zenhub.com/workspaces/notify-planning-614b3ad91bc2030015ed22f5/issues/gh/cds-snc/notification-planning/725). This is part of a larger effort to detect realtime misuse of Notify, which was a recommendation from a recent security audit.
+
+The SRE team has created the [scan-files](https://github.com/cds-snc/scan-files) project, which is a tool to help platform products scan files for malicious content. It includes terraform for [S3 scan object](https://github.com/cds-snc/scan-files/tree/main/module/s3-scan-object), which is sets up a lambda function that watches a given S3 bucket and scans each new file. The scan verdict is then written as a tag to the S3 object, and can be read when the S3 object is accessed.
+
+We have [tried getting the S3 scan object lambda function to run on our S3 bucket](https://github.com/cds-snc/notification-terraform/pull/600), but that did not work because we are using a different encryption key for each file in S3. This is because our S3 bucket for file uploads is public, and we need to prevent a user from being able to download and decrypt files they should not have access to.
```

Comment:
We checked during the meeting if our S3 bucket was public and our assumption was wrong: it's private. We actually deliver link from dd-api as a proxy. Someone who would want to download a file he shouldn't have access would have to guess the S3 file UUID and the encryption key.